cmd/telemetrygen/internal/logs/logs.go

```diff
@@ -82,3 +84,24 @@ func Run(c *Config, exp exporter, logger *zap.Logger) error {
 	wg.Wait()
 	return nil
 }
+
+func parseSeverity(severityText string) (severityNumber plog.SeverityNumber) {
```

Comment:
I am confused about this: the CLI flag talks about severity text, but here we convert to severity number. These two fields are both optional and one or the other could be set (although there is some relation between them, see https://opentelemetry.io/docs/specs/otel/logs/data-model/#severity-fields for the full details).

Which one are you interested in adding? Is it severity text, severity number, or both? If it is severity text, then you should set severity text directly, not convert to severity number

Reply:
I want to add both of them. I found that there are standard severity text and severity number:
https://github.com/open-telemetry/opentelemetry-collector/blob/fa1b24184cb20ed0584457ea68c49c908da5f988/pdata/plog/severity_number.go#L42-L55

Although both fields are optional, should we refer to the definition of severity_number.go while setting them?

if we should, the `parseSeverity` func is something like a standardization procedure,  `parseSeverity` converts the CLI flag input severity text, which may be written as `info` instead of the standard `Info`, into the standard severity number which can render the standard severity text.

Reply:
> I want to add both of them. I found that there are standard severity text and severity number:
[open-telemetry/opentelemetry-collector@fa1b241/pdata/plog/severity_number.go#L42-L55](https://github.com/open-telemetry/opentelemetry-collector/blob/fa1b24184cb20ed0584457ea68c49c908da5f988/pdata/plog/severity_number.go#L42-L55)
> Although both fields are optional, should we refer to the definition of severity_number.go while setting them?

That comes from https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity. This is a way to display the severity number, but there is (at least the way I understand it) no explicit relation between this and severity text/

I think we should just allow users to set both fields independently and arbitrarily, with the only restriction that severity number must be in the range 1-24. If any, a restriction that we could impose is that if severity text has a well-known value (e.g. `INFO`), then severity number should match it (e.g. in the case of `INFO`, it should be within the range 9-12).

Reply:
> I want to add both of them. I found that there are standard severity text and severity number:
[open-telemetry/opentelemetry-collector@fa1b241/pdata/plog/severity_number.go#L42-L55](https://github.com/open-telemetry/opentelemetry-collector/blob/fa1b24184cb20ed0584457ea68c49c908da5f988/pdata/plog/severity_number.go#L42-L55)
> Although both fields are optional, should we refer to the definition of severity_number.go while setting them?

That comes from https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity. This is a way to display the severity number, but there is (at least the way I understand it) no explicit relation between this and severity text/

I think we should just allow users to set both fields independently and arbitrarily, with the only restriction that severity number must be in the range 1-24. If any, a restriction that we could impose is that if severity text has a well-known value (e.g. `INFO`), then severity number should match it (e.g. in the case of `INFO`, it should be within the range 9-12).

Reply:
> I want to add both of them. I found that there are standard severity text and severity number:
[open-telemetry/opentelemetry-collector@fa1b241/pdata/plog/severity_number.go#L42-L55](https://github.com/open-telemetry/opentelemetry-collector/blob/fa1b24184cb20ed0584457ea68c49c908da5f988/pdata/plog/severity_number.go#L42-L55)
> Although both fields are optional, should we refer to the definition of severity_number.go while setting them?

That comes from https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity. This is a way to display the severity number, but there is (at least the way I understand it) no explicit relation between this and severity text/

I think we should just allow users to set both fields independently and arbitrarily, with the only restriction that severity number must be in the range 1-24. If any, a restriction that we could impose is that if severity text has a well-known value (e.g. `INFO`), then severity number should match it (e.g. in the case of `INFO`, it should be within the range 9-12).